chirp/chirp.go

```diff
@@ -89,7 +88,20 @@ func (b *BIRDClient) exec(cmd string, args ...interface{}) (string, error) {
 	return b.readResponse()
 }
 
-var respCodeRegex = regexp.MustCompile(`^\d{4}[ -]`)
+// hasResponseCode reports whether the provided byte slice is
+// prefixed with the BIRD response code.
+// Equivalent regex: `^\d{4}[ -]`.
+func hasResponseCode(s []byte) bool {
+	if len(s) < 5 {
+		return false
+	}
+	for i := 0; i < 4; i++ {
+		if s[i] < '0' || '9' < s[i] {
```

Comment:
I think the normal Go style is the other way around, with the boundaries on the outside:

```go
   for _, b := range s[:4] {
         if '0' <= b && b <= '9' {
              continue
         }
         return false
    }
```

Reply:
Done